src/function/matrix/eigs.js

```diff
@@ -27,352 +47,170 @@ export const createEigs = /* #__PURE__ */ factory(name, dependencies, ({ typed,
-    }
-    // Modify Hij
-    Hij[i][i] = Aii
-    Hij[j][j] = Ajj
-    Hij[i][j] = 0
-    Hij[j][i] = 0
-    // 0  to i
-    for (let k = 0; k < N; k++) {
-      if (k !== i && k !== j) {
-        Hij[i][k] = Aki[k]
-        Hij[k][i] = Aki[k]
-        Hij[j][k] = Akj[k]
-        Hij[k][j] = Akj[k]
-      }
-    }
-    return Hij
+  /** @return {boolean} */
+  function isReal(arr, N, prec) {
+    for (let i = 0; i < N; i++)
+    for (let j = 0; j < N; j++)
+      // TODO proper comparison of bignum and frac
+      if ( larger( bignumber(abs(im(arr[i][j]))), prec) )
+        return false
+
+    return true
   }
 
-  // update matrix
-  function x1 (Hij, theta, i, j) {
-    const N = Hij.length
-    const c = Math.cos(theta)
-    const s = Math.sin(theta)
-    const c2 = c * c
-    const s2 = s * s
-    const Aki = createArray(N, 0)
-    const Akj = createArray(N, 0)
-    //  Aii
-    const Aii = c2 * Hij[i][i] - 2 * c * s * Hij[i][j] + s2 * Hij[j][j]
-    const Ajj = s2 * Hij[i][i] + 2 * c * s * Hij[i][j] + c2 * Hij[j][j]
-    // 0  to i
-    for (let k = 0; k < N; k++) {
-      Aki[k] = c * Hij[i][k] - s * Hij[j][k]
-      Akj[k] = s * Hij[i][k] + c * Hij[j][k]
-    }
-    // Modify Hij
-    Hij[i][i] = Aii
-    Hij[j][j] = Ajj
-    Hij[i][j] = 0
-    Hij[j][i] = 0
-    // 0  to i
-    for (let k = 0; k < N; k++) {
-      if (k !== i && k !== j) {
-        Hij[i][k] = Aki[k]
-        Hij[k][i] = Aki[k]
-        Hij[j][k] = Akj[k]
-        Hij[k][j] = Akj[k]
-      }
-    }
-    return Hij
+
+  function coerceReal(arr, N) {
+    for (let i = 0; i < N; i++)
+    for (let j = 0; j < N; j++)
+      arr[i][j] = re(arr[i][j])
   }
 
-  // get max off-diagonal value from Upper Diagonal
-  function getAij (Mij) {
-    const N = Mij.length
-    let maxMij = 0
-    let maxIJ = [0, 1]
-    for (let i = 0; i < N; i++) {
-      for (let j = i + 1; j < N; j++) {
-        if (Math.abs(maxMij) < Math.abs(Mij[i][j])) {
-          maxMij = Math.abs(Mij[i][j])
-          maxIJ = [i, j]
-        }
-      }
+
+  /** @return {'number' | 'BigNumber' | 'Complex'} */
+  function coerceTypes(mat, arr, N) {
+
+    /** @type {string} */
+    let type = mat.datatype()
+
+    if (type == 'number' || type == 'BigNumber' || type == 'Complex') {
+      return type
     }
-    return [maxIJ, maxMij]
-  }
 
-  // get max off-diagonal value from Upper Diagonal
-  function getAijBig (Mij) {
-    const N = Mij.length
-    let maxMij = 0
-    let maxIJ = [0, 1]
-    for (let i = 0; i < N; i++) {
-      for (let j = i + 1; j < N; j++) {
-        if (abs(maxMij) < abs(Mij[i][j])) {
-          maxMij = abs(Mij[i][j])
-          maxIJ = [i, j]
-        }
-      }
+    let hasNumber  = false
+    let hasBig     = false
+    let hasComplex = false
+
+    for (let i = 0; i < N; i++)
+    for (let j = 0; j < N; j++)
+    {
+      const el = arr[i][j]
+
+      if (isNumber(el) || isFraction(el))
+        hasNumber = true
```

Comment:
I personally prefer to _always_ use braces `{...}` to prevent any future bugs

Reply:
Weird fit with the no-semicolon rule. And it's objectively worse for nested for-loops. But I'm not here to bikeshed, so I refactored the code to pass `yarn lint`.